api/provider/aws/task_list.go

```diff
@@ -0,0 +1,75 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (t *TaskProvider) List() ([]*models.TaskSummary, error) {
+	clusterNames, err := listClusterNames(t.AWS.ECS, t.Config.Instance())
+
+	if err != nil {
+		return nil, err
+	}
+
+	summaries := []*models.TaskSummary{}
+	for _, clusterName := range clusterNames {
+		runningTaskARNs, err := t.listTasksByStatus(clusterName, "RUNNING")
+		stoppedTaskARNs, err := t.listTasksByStatus(clusterName, "STOPPED")
+
+		if err != nil {
+			return nil, err
+		}
+
+		environmentID := delLayer0Prefix(t.Config.Instance(), clusterName)
+
+		for _, taskARN := range append(runningTaskARNs, stoppedTaskARNs...) {
+			summary := models.TaskSummary{
+				TaskID:        *taskARN,
+				EnvironmentID: environmentID,
+			}
+			summaries = append(summaries, &summary)
+		}
+
+	}
+
+	if err := t.populateTaskSummaries(summaries); err != nil {
+		return nil, err
+	}
+
+	return summaries, nil
+
+}
+
+func (t *TaskProvider) listTasksByStatus(clusterName, status string) ([]*string, error) {
+	input := &ecs.ListTasksInput{}
+	input.SetCluster(clusterName)
+	input.SetDesiredStatus(status)
+
+	tasks, err := t.AWS.ECS.ListTasks(input)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return tasks.TaskArns, nil
+}
+
+func (t *TaskProvider) populateTaskSummaries(summaries []*models.TaskSummary) error {
+	tags, err := t.TagStore.SelectByType("task")
+	if err != nil {
+		return err
+	}
+
+	for i, summary := range summaries {
+		if tag, ok := tags.WithID(summary.TaskID).WithKey("task_name").First(); ok {
```

Comment:
key is `name`, we also don't use an `environment_name` key. 
You should be able to copy+paste the logic from [here](https://github.com/quintilesims/layer0/blob/api-refactor/api/provider/aws/load_balancer_list.go#L53)